shared/src/commonMain/kotlin/kmp/shared/extension/Store.kt

```diff
@@ -0,0 +1,70 @@
+package kmp.shared.extension
+
+import kmp.shared.base.ErrorResult
+import kmp.shared.base.ErrorResultException
+import kmp.shared.base.Result
+import kmp.shared.base.error.domain.CommonError
+import kmp.shared.infrastructure.model.UserDto
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNot
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import org.mobilenativefoundation.store.store5.MutableStore
+import org.mobilenativefoundation.store.store5.StoreReadRequest
+import org.mobilenativefoundation.store.store5.StoreReadResponse
+import org.mobilenativefoundation.store.store5.StoreWriteResponse
+
+fun <Key : Any, Output : Any> MutableStore<Key, Output>.getResultFlow(key: Key): Flow<Result<Output>> =
+    stream<UserDto>(StoreReadRequest.cached(key, refresh = true))
+        .filterNot { it is StoreReadResponse.Loading || it is StoreReadResponse.NoNewData }
+        .map(StoreReadResponse<Output>::toResult)
+
+suspend fun <Key : Any, Output : Any> MutableStore<Key, Output>.getResult(key: Key): Result<Output> =
+    getResultFlow(key).first()
+
+inline fun <T : Any> StoreReadResponse<T>.toResult(): Result<T> = when (this) {
+    is StoreReadResponse.Data -> Result.Success(value)
+
+    is StoreReadResponse.Error.Exception -> Result.Error(CommonError.UnhandledExceptionError(error))
+    is StoreReadResponse.Error.Message -> Result.Error(
+        CommonError.UnhandledExceptionError(RuntimeException(message)),
+    )
+
+    is StoreReadResponse.Error.Custom<*> -> {
+        val errorResult: ErrorResult = error as? ErrorResult ?: error("Unknown custom error $error")
```

Comment:
Maybe defaulting to an unknown error would be better than crashing the app?

Reply:
Yeah, that's an endless discussion. I'd say it's better to crash than have a clear bug in your code. If you pass anything else than ErrorResult as a custom error, nothing else will ever be able to read it and it will for ever stay as a unknown error.